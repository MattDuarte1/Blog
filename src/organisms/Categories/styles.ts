import styled, { css, DefaultTheme } from 'styled-components';

const BreakPoints = {
  ContainerSize: {
    medium: (theme: DefaultTheme) => css`
      @media ${theme.media.lteMedium} {
        padding: ${theme.spacings.medium};
      }
    `,
  },
  CategoryBodySize: {
    small: (theme: DefaultTheme) => css`
      @media ${theme.media.lteSmall} {
        div {
          width: 100%;
        }
      }
    `,
  },
};

export const Container = styled.section`
  ${({ theme }) => css`
    display: block;
    padding: ${theme.spacings.xxlarge};
    background-color: ${theme.colors.white};
    width: 100%;
    height: 100%;

    ${BreakPoints.ContainerSize['medium'](theme)}
  `}
`;

export const CategoryHeader = styled.div`
  display: flex;
  justify-content: center;
  margin-bottom: 2rem;
`;

export const CategoryBody = styled.section`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row wrap;
    justify-content: center;
    min-width: 100%;
    height: 100%;
    gap: 37px;

    ${BreakPoints.CategoryBodySize['small'](theme)}
  `}
`;
