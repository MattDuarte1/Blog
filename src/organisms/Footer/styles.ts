import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    width: 100%;
    padding: 5rem;

    box-shadow: 1px 0 1rem 0 rgba(0, 0, 0, 0.2);

    display: flex;

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      gap: 4rem;
      padding: 5rem;
    }
  `}
`;

export const Content = styled.aside`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
    gap: ${theme.spacings.small};

    flex: 1;
  `}
`;

export const SocialMediaContent = styled.div`
  ${({ theme }) => css`
    display: flex;
    gap: 2rem;
    justify-content: center;
    width: 40%;
    a {
      color: ${theme.colors.black};
    }

    text-decoration: none;
    color: ${theme.colors.black};
    svg {
      cursor: pointer;
      font-size: 25px;
      transition: all 0.4s ease;

      &:hover {
        transform: scale(1.1);
      }
    }
  `}
`;
