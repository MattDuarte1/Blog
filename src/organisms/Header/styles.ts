import styled, { css } from 'styled-components';

export const Container = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-around;
    width: 100%;
    height: 86px;
    background-color: ${theme.colors.white};
    filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.05));

    position: relative;
  `}
`;
