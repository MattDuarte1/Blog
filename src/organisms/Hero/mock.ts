export default {
  title: 'Bem vindo ao blog',
  description: 'Blog Desenvolvido por Mateus Duarte',
  image: {
    url: 'https://www.datocms-assets.com/90479/1672435055-group.svg',
    alt: 'alt imagem',
    custom: {
      width: '350',
      height: '350',
    },
  },
};
