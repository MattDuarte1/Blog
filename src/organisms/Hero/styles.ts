import styled, { css, DefaultTheme, keyframes } from 'styled-components';

const BreakPoints = {
  Container: {
    medium: (theme: DefaultTheme) => css`
      @media ${theme.media.lteLarge} {
        padding: ${theme.spacings.huge} ${theme.spacings.medium};
      }
    `,
  },
  DescriptionSize: {
    small: (theme: DefaultTheme) => css`
      @media ${theme.media.lteSmall} {
        margin-top: 20px;
        font-size: ${theme.font.sizes.xsmall};
        line-height: normal;
      }
    `,
    medium: (theme: DefaultTheme) => css`
      @media ${theme.media.lteMedium} {
        margin-top: 40px;
        font-size: ${theme.font.sizes.small};
        line-height: normal;
      }
    `,
  },
  AsideRight: {
    medium: (theme: DefaultTheme) => css`
      @media ${theme.media.lteMedium} {
        img {
          width: 280px;
          height: 280px;
        }
      }
    `,
  },
};

const cursorEffect = (theme: DefaultTheme) => keyframes`
  0% {
    background-color: ${theme.colors.black};
  }
  100% {
    background-color: ${theme.colors.white};
  }
`;

export const Container = styled.section`
  ${({ theme }) => css`
    width: 100%;
    padding: ${theme.spacings.xhuge} ${theme.spacings.medium};
    min-height: 700px;
    display: flex;
    background-color: ${theme.colors.white};
    align-items: center;
    justify-content: center;
    flex-wrap: wrap;
    gap: 20px;

    ${BreakPoints.Container['medium'](theme)}
  `}
`;

export const AsideLeft = styled.aside`
  ${({ theme }) => css`
    max-height: 400px;
    max-width: 664px;
    padding: 25px
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    gap: 30px;

    span {
      color: transparent;
      width: 5px;
      height: 100%;
      width: 2px;
      animation: ${cursorEffect(theme)} 1s infinite;
    }
  `}
`;

export const AsideRight = styled(AsideLeft)`
  ${({ theme }) => css`
    justify-content: center;
    display: block;

    ${BreakPoints.AsideRight['medium'](theme)}
  `}
`;

export const Description = styled.p`
  ${({ theme }) => css`
    padding-left: 15px;
    margin-top: 25px;
    font-family: ${theme.font.family.default};
    font-style: normal;
    font-weight: 400;
    font-size: ${theme.font.sizes.medium};
    color: ${theme.colors.darkGrey};
    line-height: 25px;
    margin-left: 20px;
    position: relative;

    &::before {
      content: '';
      position: absolute;
      left: 0px;
      background-color: ${theme.colors.black};
      width: 5px;
      height: 100%;
    }

    ${BreakPoints.DescriptionSize['small'](theme)}
    ${BreakPoints.DescriptionSize['medium'](theme)}
  `}
`;
