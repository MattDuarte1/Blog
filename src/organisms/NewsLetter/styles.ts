import styled, { css, DefaultTheme } from 'styled-components';

const BreakPoints = {
  Content: (theme: DefaultTheme) => css`
    @media ${theme.media.lteMedium} {
      > h1 {
        font-size: ${theme.font.sizes.medium};
      }

      @media ${theme.media.lteSmall} {
        > h1 {
          font-size: ${theme.font.sizes.small};
        }
        > span {
          font-size: ${theme.font.sizes.xsmall};
        }
      }
    }
  `,
};

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 58px;
    gap: 10px;
    background-color: ${theme.name === 'inverted'
      ? theme.colors.white
      : theme.colors.lightGrey2};
    width: 100%;
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 0px;
    gap: 30px;

    max-width: 696px;

    ${BreakPoints.Content(theme)}
  `}
`;

export const Form = styled.form`
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
  padding: 0px;
  gap: 20px;
  width: 100%;
`;

export const InputForm = styled.input`
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 10px;
  padding: 21px 21px 21px 25px;

  width: 300px;
  height: 67px;

  /* White */

  background: #fffffe;
  border-radius: 8px;
  border: none;

  &:focus {
    box-shadow: 0 0 0 0;
    outline: 0;
  }
`;
