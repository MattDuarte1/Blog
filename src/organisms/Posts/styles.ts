import styled, { css, DefaultTheme } from 'styled-components';

const MediaBreakPoints = {
  small: (theme: DefaultTheme) => css`
    @media ${theme.media.lteSmall} {
      padding: 3.5rem;
    }
  `,
};

export const Container = styled.div`
  ${({ theme }) => css`
    padding: 4.4rem;
    background-color: ${theme.colors.white};
    ${MediaBreakPoints.small(theme)}
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    margin-top: 1.4rem;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(26rem, 2fr));
    gap: ${theme.spacings.large};

    @media (max-width: 1065px) {
      grid-template-columns: repeat(auto-fill, minmax(27rem, 1fr));
    }

    @media (max-width: 705px) {
      justify-content: center;
      grid-template-columns: 0fr;
    }
  `}
`;
