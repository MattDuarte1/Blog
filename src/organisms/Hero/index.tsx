import Image from 'next/image';
import { IHero } from '@/interfaces/Hero';
import * as Styled from './styles';
import Heading from '@/atoms/Heading';
import { forwardRef } from 'react';

type HeroProps = {
  data?: IHero;
};

const Hero = forwardRef<HTMLDivElement, HeroProps>(
  ({ data, ...props }, ref) => {
    const width = Number(data?.image.custom.width);
    const height = Number(data?.image.custom.height);

    return (
      <Styled.Container id="hero" ref={ref} {...props}>
        <Styled.AsideLeft>
          <Heading color="black" size={'xxlarge'}>
            {data?.title}
          </Heading>
          <Styled.Description>{data?.description}</Styled.Description>
        </Styled.AsideLeft>
        <Styled.AsideRight>
          <Image
            src={data?.image.url as string}
            width={width}
            priority={true}
            height={height}
            alt={`${data?.image.alt}`}
          />
        </Styled.AsideRight>
      </Styled.Container>
    );
  },
);

Hero.displayName = 'Hero Component';

export default Hero;
