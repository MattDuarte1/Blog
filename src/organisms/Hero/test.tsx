import { screen } from '@testing-library/react';
import Hero from '.';
import { renderTheme } from '@/styles/render-theme';
import { theme } from '@/styles/theme';
import mockHero from './mock';
import React from 'react';

describe('<Hero />', () => {
  it('Should render correctly hero section', async () => {
    const { container } = renderTheme(<Hero data={mockHero} />);
    const description = container.querySelector('p');
    const imagem = screen.getByRole('img');

    expect(description).toBeInTheDocument();
    expect(imagem).toBeInTheDocument();
    expect(imagem).toHaveStyle({
      width: mockHero.image.custom.width,
      height: mockHero.image.custom.height,
    });
  });

  it('Should render correctly hero section with breakpoints screen is lteMedium', () => {
    const { container } = renderTheme(<Hero data={mockHero} />);
    const description = container.querySelector('p');

    // description
    expect(description).toHaveStyleRule('margin-top', '40px', {
      media: theme.media.lteMedium,
    });
    expect(description).toHaveStyleRule('font-size', theme.font.sizes.small, {
      media: theme.media.lteMedium,
    });
    expect(description).toHaveStyleRule('line-height', 'normal', {
      media: theme.media.lteMedium,
    });
  });

  it('Should render correctly hero section with breakpoints screen is ltSmall', () => {
    const { container } = renderTheme(<Hero data={mockHero} />);
    const description = container.querySelector('p');

    // description
    expect(description).toHaveStyleRule('margin-top', '20px', {
      media: theme.media.lteSmall,
    });

    expect(description).toHaveStyleRule('font-size', theme.font.sizes.xsmall, {
      media: theme.media.lteSmall,
    });

    expect(description).toHaveStyleRule('line-height', 'normal', {
      media: theme.media.lteSmall,
    });
  });

  it('Should match snapshot component', () => {
    const { container } = renderTheme(<Hero data={mockHero} />);

    expect(container).toMatchSnapshot();
  });
});
