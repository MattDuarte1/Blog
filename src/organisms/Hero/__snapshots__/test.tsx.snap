// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> Should match snapshot component 1`] = `
.c0 {
  width: 100%;
  padding: 28.8rem 2.4rem;
  min-height: 700px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #FFFFFE;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  gap: 20px;
}

.c1 {
  max-height: 400px;
  max-width: 664px;
  padding: 25px display:flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  gap: 30px;
}

.c1 span {
  color: transparent;
  width: 5px;
  height: 100%;
  width: 2px;
  -webkit-animation: gqltWr 1s infinite;
  animation: gqltWr 1s infinite;
}

.c4 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: block;
}

.c3 {
  padding-left: 15px;
  margin-top: 25px;
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 400;
  font-size: 2.4rem;
  color: #9A9494;
  line-height: 25px;
  margin-left: 20px;
  position: relative;
}

.c3::before {
  content: '';
  position: absolute;
  left: 0px;
  background-color: #2B2C34;
  width: 5px;
  height: 100%;
}

.c2 {
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 700;
  line-height: 1.2;
  text-transform: capitalize;
  font-size: 6.4rem;
  text-transform: none;
  color: #2B2C34;
}

@media (max-width:999px) {
  .c0 {
    padding: 10.06rem 2.4rem;
  }
}

@media (max-width:700px) {
  .c4 img {
    width: 280px;
    height: 280px;
  }
}

@media (max-width:400px) {
  .c3 {
    margin-top: 20px;
    font-size: 1.617rem;
    line-height: normal;
  }
}

@media (max-width:700px) {
  .c3 {
    margin-top: 40px;
    font-size: 2.156rem;
    line-height: normal;
  }
}

@media (max-width:700px) {
  .c2 {
    font-size: 4.8rem;
  }
}

@media (max-width:400px) {
  .c2 {
    font-size: 3.576rem;
  }
}

<div>
  <section
    class="c0"
    id="hero"
  >
    <aside
      class="c1"
    >
      <h1
        class="c2"
        color="black"
      >
        Bem vindo ao blog
      </h1>
      <p
        class="c3"
      >
        Blog Desenvolvido por Mateus Duarte
      </p>
    </aside>
    <aside
      class="c1 c4"
    >
      <img
        alt="alt imagem"
        data-nimg="1"
        decoding="async"
        height="350"
        src="https://www.datocms-assets.com/90479/1672435055-group.svg"
        srcset="https://www.datocms-assets.com/90479/1672435055-group.svg 1x, https://www.datocms-assets.com/90479/1672435055-group.svg 2x"
        style="color: transparent;"
        width="350"
      />
    </aside>
  </section>
</div>
`;
